codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithygo/awssdk/DafnyAwsSdkClientTypeConversionProtocol.java

```diff
@@ -588,23 +588,28 @@ func OpaqueError_Input_ToDafny(nativeInput error)($L.Error) {
           writer.write(
             """
             func Error_ToDafny(err error)($L.Error) {
-                switch err.(type) {
                 // Service Errors
                 ${C|}
-
-                default:
-                    return OpaqueError_Input_ToDafny(err)
-                }
+                return OpaqueError_Input_ToDafny(err)
             }
             """,
             DafnyNameResolver.dafnyTypesNamespace(serviceShape),
             writer.consumer(w -> {
               for (final var error : errorShapes) {
+                final String errVariableName = context
+                .symbolProvider()
+                .toSymbol(
+                  awsNormalizedModel.expectShape(error.toShapeId())
+                ).getName();
+                w.addImport("errors");
                 w.write(
                   """
-                    case *$L:
-                        return $L(*err.(*$L))
+                    var $L *$L
+                    if (errors.As(err , &$L)) {
```

Comment:
having a if block looks more readable instead of a switch in here. 
example of if statement:
```
{
      var CloudHsmClusterNotRelatedException *types.CloudHsmClusterNotRelatedException
      if (errors.As(err , &CloudHsmClusterNotRelatedException)) {
        return CloudHsmClusterNotRelatedException_ToDafny(*CloudHsmClusterNotRelatedException)
      }

      var InvalidGrantIdException *types.InvalidGrantIdException
      if (errors.As(err , &InvalidGrantIdException)) {
        return InvalidGrantIdException_ToDafny(*InvalidGrantIdException)
      }

      var TagException *types.TagException
      if (errors.As(err , &TagException)) {
        return TagException_ToDafny(*TagException)
      }
     ...
     return opaque...
}
```

for switch case it would be like 
```
{
     var CloudHsmClusterNotRelatedException *types.CloudHsmClusterNotRelatedException
     var InvalidGrantIdException *types.InvalidGrantIdException
     var TagException *types.TagException
     (at least around 40 other var initialization)
     
     switch (...) {
     
     }
}
```

Reply:
Also, errors.As is being used and this can't directly changed to switch

Reply:
Ahh the variable initialization.